### ISSUE TITLE
Error "error[E0603]: module `xxx` is private" when compiling

### ISSUE THREAD

**Issue by tbontb-iaq** [daboross/rust-screeps-api#12]
I just add `screeps-api` to my project, and when I compile, this error occurs:

```text
> Executing task: cargo run --package screeps --bin screeps <

   Compiling screeps-api v0.6.0
error[E0603]: module `export` is private
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/screeps-api-0.6.0/src/websocket/types/room/room_object_macros.rs:64:65
    |
64  |             fn deserialize<__D>(__deserializer: __D) -> _serde::export::Result<Self, __D::Error>
    |                                                                 ^^^^^^ private module
    |
note: the module `export` is defined here
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.137/src/lib.rs:274:5
    |
274 | use self::__private as export;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: module `private` is private
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/screeps-api-0.6.0/src/websocket/types/room/room_object_macros.rs:71:30
    |
71  |                     <_serde::private::de::Content as Deserialize>::deserialize(__deserializer)?;
    |                              ^^^^^^^ private module
    |
note: the module `private` is defined here
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.137/src/lib.rs:276:5
    |
276 | use self::__private as private;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: module `private` is private
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/screeps-api-0.6.0/src/websocket/types/room/room_object_macros.rs:74:33
    |
74  |                         _serde::private::de::ContentRefDeserializer::<__D::Error>::new(&__content),
    |                                 ^^^^^^^ private module
    |
note: the module `private` is defined here
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.137/src/lib.rs:276:5
    |
276 | use self::__private as private;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: module `private` is private
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/screeps-api-0.6.0/src/websocket/types/room/room_object_macros.rs:83:33
    |
83  |                         _serde::private::de::ContentRefDeserializer::<__D::Error>::new(&__content),
    |                                 ^^^^^^^ private module
    |
note: the module `private` is defined here
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.137/src/lib.rs:276:5
    |
276 | use self::__private as private;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: module `export` is private
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/screeps-api-0.6.0/src/websocket/types/room/room_object_macros.rs:90:25
    |
90  |                 _serde::export::Err(_serde::de::Error::custom(format!(
    |                         ^^^^^^ private module
    |
note: the module `export` is defined here
   --> /home/jin/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.137/src/lib.rs:274:5
    |
274 | use self::__private as export;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0603`.
error: could not compile `screeps-api` due to 5 previous errors
```

And this is my Cargo.toml file:

```toml
[package]
name = "screeps"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
[profile.release]
opt-level = 'z'
lto = true
codegen-units = 1
panic = 'abort'
strip = true

[dependencies]
serde = { version = "1", features = ["derive"] }
lazy_static = "1.4"
serde_json = "1.0"
env_logger = "0.9"
log = "0.4"

actix-web = { version = "4", features = ["openssl"] }
openssl = { version = "0.10", features = ["vendored"] }
screeps-api = "0.6.0"

reqwest = { version = "0.11", features = ["json"] }
tokio = { version = "1", features = ["full"] }      # async
reqwest-middleware = "0.1"
reqwest-retry = "0.1"
base64 = "0.13"
flate2 = "1.0"                                      # Gzip
```

**Comment by daboross** [comment]
Looks like this was already fixed in master with 65f6da415075cc67a4597bed575cc2179a46de8c. 

I recommend depending on the git release version with:

```
screeps-api = { git = "https://github.com/daboross/rust-screeps-api" }
```

I may release a fixed version, but it's less of a priority for me right now. If a fixed version published on crates.io would be helpful for you, though, let me know & I'll do that release process.

**Comment by tbontb-iaq** [comment]
Thanks a lot, my problem is solved.
